tests/Validity.hs

```diff
@@ -270,6 +271,26 @@ pNull = producesValidsOnValids (HM.null :: HM.HashMap Key Int -> Bool)
 pSize :: Property
 pSize = producesValidsOnValids (HM.size :: HM.HashMap Key Int -> Int)
 
+pEqual1 :: Fun (Int, Int) Bool -> Property
+pEqual1 f =
+    producesValidsOnValids2
+        (HM.equal1 (applyFun2 f) :: HM.HashMap Key Int -> HM.HashMap Key Int -> Bool)
```

Comment:
These makes no sense to me whatsoever. Every `Bool` is inherently valid and every `Ordering` is inherently valid. What are you trying to accomplish? I don't think we need these tests at all. We *do* want to add more detailed `Eq2` and `Ord2` tests at some point, but those are going to be on the subtle side. @phadej has promised an explanation of what preconditions those should have, and I'm going to hold him to it. I still hate the way those classes work/not for dictionaries and sets.